types/use-double-click/index.d.ts

```diff
@@ -0,0 +1,17 @@
+// Type definitions for use-double-click 1.0
+// Project: https://github.com/tim-soft/use-double-click
+// Definitions by: Matthew Peveler <https://github.com/MasterOdin>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { RefObject, MutableRefObject, MouseEvent } from 'react';
+
+export default function useDoubleClick<T = unknown>(options: {
+  /** Dom node to watch for double clicks */
+  ref: RefObject<T> | MutableRefObject<T>;
+  /** The amount of time (in milliseconds) to wait before differentiating a single from a double click. Defaults to 300. */
+  latency?: number;
+  /** A callback function for single click events */
+  onSingleClick?: (e: MouseEvent) => void;
+  /** A callback function for double click events */
+  onDoubleClick?: (e: MouseEvent) => void;
+}): void;
```

Comment:
This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like in the following:

```ts
class Foo {
  // ...
}

export = Foo
```

To import the entity using a default import, you will need to enable `esModuleInterop`. Alternatively, you can just import using a more appropriate syntax like

```ts
import foo = require("foo");
```

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).